File: rb/features.rb
# UselessFacts SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module UselessFactsFeatures
  def self.make_feature(name)
    case name
    when "base"
      UselessFactsBaseFeature.new
    when "test"
      UselessFactsTestFeature.new
    else
      UselessFactsBaseFeature.new
    end
  end
end

File: py/features.py
# UselessFacts SDK feature factory

from feature.base_feature import UselessFactsBaseFeature
from feature.test_feature import UselessFactsTestFeature


def _make_feature(name):
    features = {
        "base": lambda: UselessFactsBaseFeature(),
        "test": lambda: UselessFactsTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
